control-plane/api/v1alpha1/serviceresolver_types.go

```diff
@@ -522,27 +485,35 @@ func (in ServiceResolverFailoverMap) toConsul() map[string]capi.ServiceResolverF
 	}
 	m := make(map[string]capi.ServiceResolverFailover)
 	for k, v := range in {
-		m[k] = v.toConsul()
+		m[k] = *v.toConsul()
```

Comment:
Consul Core checks if redirect or failover values are empty and if so it errors out, also both failover and redirect can't be set at the same time. I think it's better to return a nil rather than an empty struct here. But I will add a line to check for nil values. Thanks for catching this.

Reply:
resolving this convo: addressed this comment.